### PR TITLE
Add heading offset to Point At Constraint definition

### PR DESCRIPTION
**Pull request by jlmcmchl**
This PR adds support for a heading offset to the Point At constraint.

subtracting heading offset from the robot angle results in positive heading offset leading to the robot being "more counterclockwise" relative to the point it should face, and vice-versa for negative heading offset. This felt right because heading increases in the counterclockwise direction, but if the other way would be more intuitive that's cool too.

I've noticed that the point-at constraint really wants to be included in the initial guess, as applying this constraint to translation waypoints that are facing the wrong direction quickly leads to infeasibility. It's not immediately clear how to accomplish this, but when applying this constraint it might be useful to apply it to the initial guesses at each waypoint, though we might be able to get better results if we can override the initial guess' heading with where the constraint says it should face.

### CODE REVIEW

**Comment by calcmogul** [comment]
CCW-positive offset sounds good.

File: src/lib.rs [hunk 1]
```diff
@@ -149,6 +150,7 @@ mod ffi {
             field_point_x: f64,
             field_point_y: f64,
             heading_tolerance: f64,
+            offset: f64,
```
Comment (bruingineer) [inline src/lib.rs:153]: `heading_offset`

**Comment by calcmogul** [comment]
The merge conflict here is uninformative, so the fix is to add the heading offset argument to some existing functions in include/trajopt/constraint/PointAtConstraint.hpp, src/trajoptlibrust.cpp, and src/trajoptrust.hpp.

**Comment by calcmogul** [comment]
After further discussion on the Discord, we noted that this feature is a footgun for users if they don't use an offset of 0° or 180°. The mirrored trajectory won't point at the correct place. To get the correct trajectory, you have to invert the offset and rerun the optimization.

We're open to making the point-at constraint toggle between front and back pointing (the 0° and 180° cases mentioned earlier).

**Comment by calcmogul** [comment]
TrajoptLib has moved to a subfolder within SleipnirGroup/Choreo. The changes here should be merged into SleipnirGroup/Choreo#438.

**Pull request closed without merging**